enterprise_access/apps/customer_billing/models.py

```diff
@@ -425,28 +429,63 @@ def create_intent(
         with transaction.atomic():
             cls.cleanup_expired()
 
-            if not cls.can_reserve(slug, name, exclude_user=user):
-                raise ValueError(f"Slug '{slug}' or name '{name}' is already reserved")
+            if bool(slug) != bool(name):
+                raise ValueError("slug and name must either both be given or neither be given.")
 
             existing_intent = cls.objects.filter(user=user).first()
 
-            expires_at = timezone.now() + timedelta(minutes=cls.get_reservation_duration())
-
+            # If an existing intent has already reached a terminal state, exit fast.
             if existing_intent:
                 if existing_intent.state in cls.SUCCESS_STATES:
                     return existing_intent
 
                 if existing_intent.state in cls.FAILURE_STATES:
                     raise ValueError("Failed checkout record already exists")
 
+            # Establish whether or not a new slug needs to be reserved.
+            #
+            #       |    Slug    |  Intent  |      A Slug       | Requested Slug |    Requested Slug
+            #  Case | Requested? | Existed? | Already Reserved? | Is Different?  | Needs To Be Reserved
+            # ------+------------+----------+-------------------+----------------+---------------------
+            #   1   |     no     |    no    |        N/A        |      N/A       |        no
+            #   2   |     no     |    yes   |        no         |      N/A       |        no
+            #   3   |     no     |    yes   |        yes        |      N/A       |        no
+            #   4   |     yes    |    no    |        N/A        |      N/A       |        yes
+            #   5   |     yes    |    yes   |        no         |      N/A       |        yes
+            #   6   |     yes    |    yes   |        yes        |      no        |        no
+            #   7   |     yes    |    yes   |        yes        |      yes       |        yes
+            should_reserve_new_slug = False
+            if slug:
+                if existing_intent:
+                    slug_changing = slug != existing_intent.enterprise_slug
+                    name_changing = name != existing_intent.enterprise_name
+                    should_reserve_new_slug = slug_changing or name_changing  # Cases 5, 6, 7
+                else:
+                    should_reserve_new_slug = True  # Case 4
+            else:
+                pass  # Cases 1, 2, 3
```

Comment:
technicall we don't need this `else`.  maybe a comment next to `should_reserve_new_slug = False` like "default to False, covers cases # 1, 2, 3